# Standardizr

[mgrast-3](http://blog.metagenomics.anl.gov/howto/mg-rast-analysis-tools/) uses this cool technique to standardize Illumina reads per OTU data. I wrote a script that does the same.

  $ python ./standardizr.py test_data.csv
  	A	B	C	D
  i	-1.3683482773	-1.32161720241	-1.3416994809	-1.30580644296
  k	0.52446681282	0.505846996739	0.514784651002	0.497850716856
  j	-0.0763009717669	-0.16861566558	-0.131035559326	-0.196279569042
  l	0.920182436245	0.984385871248	0.957950389229	1.00423529515